libs/chonky/src/lib.rs

```diff
@@ -1,26 +1,134 @@
 #![doc = include_str!("../README.md")]
 
-/// Adds two numbers together
-///
-/// # Example
-///
-/// ```rust
-/// use chonky::add;
-///
-/// assert_eq!(add(1, 3), 4);
-/// ```
-#[must_use]
-pub const fn add(left: u64, right: u64) -> u64 {
-    left + right
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum ChonkyError {
+    #[error("parsing error in pdf")]
+    ReadPdf,
+    #[error("pdfium error")]
+    Pdfium,
+    #[error("write error to system")]
+    Write,
+    #[error("Issues with CLI input")]
+    Arguments,
+}
+
+pub mod pdf_segmentation {
+    use error_stack::{Report, ResultExt};
+    use image::DynamicImage;
+    use pdfium_render::prelude::*;
```

Comment:
We typically try to avoid glob-imports (except in tests, which often imports `super::*`). This pulls in a whole namespace, similar as you would do `std::*` in C++. `pdfium-render` seems a bit weird here because from the documentation it seems that _only_ a prelude is provided (and the prelude is quite big). Let's not use a glob-import here.